Eigen 3.4.0 compatibility fix

## Reason
`Eigen::DontAlign` seems to cause issues with `Eigen::SelfAdjointEigenSolver`
Resulting in 

```
/home/schwd/Programs/libmesh/installed/include/Eigen/src/Eigenvalues/Tridiagonalization.h: In instantiation of 'void Eigen::internal::tridiagonalization_inplace(MatrixType&, DiagonalType&, SubDiagonalType&, CoeffVectorType&, bool) [with MatrixType = Eigen::Matrix<double, 6, 6, 0, 6, 6>; DiagonalType = Eigen::Matrix<double, 6, 1, 0, 6, 1>; SubDiagonalType = Eigen::Matrix<double, 5, 1, 2, 5, 1>; CoeffVectorType = Eigen::Matrix<double, 5, 1, 2, 5, 1>]':
/home/schwd/Programs/libmesh/installed/include/Eigen/src/Eigenvalues/SelfAdjointEigenSolver.h:458:39:   required from 'Eigen::SelfAdjointEigenSolver<MatrixType>& Eigen::SelfAdjointEigenSolver<_MatrixType>::compute(const Eigen::EigenBase<OtherDerived>&, int) [with InputType = Eigen::Matrix<double, 6, 6, 2>; _MatrixType = Eigen::Matrix<double, 6, 6, 2>]'
/home/schwd/Programs/libmesh/installed/include/Eigen/src/Eigenvalues/SelfAdjointEigenSolver.h:181:14:   required from 'Eigen::SelfAdjointEigenSolver<_MatrixType>::SelfAdjointEigenSolver(const Eigen::EigenBase<OtherDerived>&, int) [with InputType = Eigen::Matrix<double, 6, 6, 2>; _MatrixType = Eigen::Matrix<double, 6, 6, 2>]'
/home/schwd/Programs/moose/modules/tensor_mechanics/src/materials/ADHillElastoPlasticityStressUpdate.C:147:59:   required from here
/home/schwd/Programs/libmesh/installed/include/Eigen/src/Eigenvalues/Tridiagonalization.h:434:55: error: no matching function for call to 'Eigen::internal::tridiagonalization_inplace_selector<Eigen::Matrix<double, 6, 6, 0, 6, 6>, 6, false>::run(Eigen::Matrix<double, 6, 6, 0, 6, 6>&, Eigen::Matrix<double, 6, 1, 0, 6, 1>&, Eigen::Matrix<double, 5, 1, 2, 5, 1>&, Eigen::Matrix<double, 5, 1, 2, 5, 1>&, bool&)'
  434 |   tridiagonalization_inplace_selector<MatrixType>::run(mat, diag, subdiag, hcoeffs, extractQ);
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Design
Remove `Eigen::DontAlign`

## Impact
Code will build with Eigen 3.4.0

--- a/framework/include/utils/RankTwoTensorImplementation.h
+++ b/framework/include/utils/RankTwoTensorImplementation.h
@@ -969,7 +969,7 @@ void
 ADRankTwoTensor::symmetricEigenvaluesEigenvectors(std::vector<ADReal> & eigvals,
                                                   ADRankTwoTensor & eigvecs) const
 {
-  typedef Eigen::Matrix<ADReal, N, N, Eigen::DontAlign> RankTwoMatrix;
+  typedef Eigen::Matrix<ADReal, N, N> RankTwoMatrix;
   RankTwoMatrix self;
   for (auto i : make_range(N))
     for (unsigned int j = i; j < N; ++j)

--- a/modules/tensor_mechanics/include/materials/ADGeneralizedRadialReturnStressUpdate.h
+++ b/modules/tensor_mechanics/include/materials/ADGeneralizedRadialReturnStressUpdate.h
@@ -34,10 +34,10 @@ struct cast_impl<ADReal, int>
 
 // Instantiation of ADReal will be enabled shortly, once
 // https://github.com/roystgnr/MetaPhysicL/issues/70 makes it to MOOSE
-// typedef Eigen::Matrix<ADReal, 6, 6, Eigen::DontAlign> AnisotropyMatrix;
+// typedef Eigen::Matrix<ADReal, 6, 6> AnisotropyMatrix;
 
-typedef Eigen::Matrix<Real, 6, 6, Eigen::DontAlign> AnisotropyMatrixReal;
-typedef Eigen::Matrix<Real, 3, 3, Eigen::DontAlign> AnisotropyMatrixRealBlock;
+typedef Eigen::Matrix<Real, 6, 6> AnisotropyMatrixReal;
+typedef Eigen::Matrix<Real, 3, 3> AnisotropyMatrixRealBlock;
 
 /**
  * ADGeneralizedRadialReturnStressUpdate computes the generalized radial return stress increment for

